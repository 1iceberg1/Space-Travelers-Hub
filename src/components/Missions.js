import React from 'react';

const Missions = () => (
  <div className="mission_container">
    <table className="table_container">
      <thead>
        <tr>
          <th>Mission</th>
          <th>Description</th>
          <th>Status</th>
          <th>Empty Title</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Thaicon</td>
          <td>dgdfhhgshkfbvfljavbfb kjfbajdfbvusdbsf sufbhalkhgvjadbvjv</td>
          <td>dgdfhhgshkfbvfljavbfb kjfbajdfbvusdbsf sufbhalkhgvjadbvjv</td>
          <td>Not a member</td>
          <td><button type="button">Join Mission</button></td>
        </tr>
      </tbody>

    </table>

  </div>

);

export default Missions;
