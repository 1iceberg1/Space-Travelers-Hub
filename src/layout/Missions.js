import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions, joinMission } from '../redux/missions/missionsSlice';
import './styles/missions.css';

const Missions = () => {
  const dispatch = useDispatch();
  const missions = useSelector((state) => state.missions.missions);

  useEffect(() => {
    if (!missions.length) {
      dispatch(fetchMissions());
    }
  }, []);

  const missionHandler = (missionId) => {
    const mission = missions.find((mission) => mission.mission_id === missionId);
    const newMission = { ...mission, joined: !mission.joined };
    const newMissions = missions.map((mission) => (mission.mission_id === missionId
      ? newMission
      : mission));
    dispatch(joinMission(newMissions));
  };

  return (
    <div className="mission_container">
      <table className="table_container">
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th>Status</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id} className="row">
              <td className="tbody_header">{mission.mission_name}</td>
              <td className="body_description pb-4">{mission.description}</td>
              <td>
                <span>{mission.joined ? 'Active member' : 'NOT A MEMBER'}</span>
              </td>
              <td>
                <button type="button" onClick={() => missionHandler(mission.mission_id)}>{mission.joined ? 'Leave mission' : 'Join mission'}</button>
              </td>
            </tr>
          ))}
        </tbody>

      </table>

    </div>

  );
};

export default Missions;
