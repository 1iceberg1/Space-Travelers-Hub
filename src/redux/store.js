import { configureStore } from '@reduxjs/toolkit';
import rocketsSlice from './rockets/rocketsSlice';
import missionsSlice from './missions/missionsSlice';

const store = configureStore({
  reducer: {
    rockets: rocketsSlice,
    missions: missionsSlice,
  },
});

export default store;
