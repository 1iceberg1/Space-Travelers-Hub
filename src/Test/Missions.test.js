import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Missions from '../layout/Missions';
import store from '../redux/store';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Missions />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
