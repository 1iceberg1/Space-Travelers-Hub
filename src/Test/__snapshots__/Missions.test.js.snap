// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="mission_container"
>
  <table
    className="table_container"
  >
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th>
          Status
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
